aten/src/ATen/native/mps/OperationUtils.mm

```diff
@@ -359,26 +370,152 @@ void printTensorNDArray(const Tensor& t) {
   return [tmpGraphTensorData mpsndarray];
 }
 
+static std::vector<int64_t> getSortedStrides(const IntArrayRef& s) {
+  std::vector<int64_t> idx(s.size());
+  iota(idx.begin(), idx.end(), 0);
+  sort(idx.begin(), idx.end(), [&s](size_t i1, size_t i2) { return s[i1] > s[i2]; });
+
+  return idx;
+}
+
+static std::vector<int64_t> inversePermutation(const std::vector<int64_t>& permuteOrder) {
+  auto size = permuteOrder.size();
+  std::vector<int64_t> inversePerm(permuteOrder.size());
+
+  for (int i = 0; i < size; i++) {
+    inversePerm[permuteOrder[i]] = i;
+  }
+  return inversePerm;
+}
+
+static MPSNDArray* permuteNDArray(MPSNDArray* inArray, const std::vector<int64_t>& permuteOrder_) {
+  auto permuteOrder = inversePermutation(permuteOrder_);
+  NSUInteger srcRank = [inArray numberOfDimensions];
+  if (srcRank != permuteOrder.size()) {
+    TORCH_INTERNAL_ASSERT(false);
+    return nil;
+  }
+  std::vector<NSUInteger> dimensionOrder(srcRank);
+  std::iota(std::begin(dimensionOrder), std::end(dimensionOrder), 0);
+  MPSNDArrayDescriptor* desc = [inArray descriptor];
+
+  for (int64_t i = srcRank - 1; i >= 0; i--) {
+    NSUInteger axis = permuteOrder[i];
+    auto axisIter = std::find(dimensionOrder.begin(), dimensionOrder.end(), axis);
+    NSUInteger axis1 = srcRank - i - 1;
+    NSUInteger axis2 = dimensionOrder.end() - axisIter - 1;
+    iter_swap(dimensionOrder.begin() + i, axisIter);
+    if (axis1 != axis2) {
+      [desc transposeDimension:axis1 withDimension:axis2];
+    }
+  }
+  C10_CLANG_DIAGNOSTIC_PUSH()
+#if C10_CLANG_HAS_WARNING("-Wnonnull")
+  C10_CLANG_DIAGNOSTIC_IGNORE("-Wnonnull")
```

Comment:
Why is this necessary?

Reply:
MPSNDArray **arrayViewWithCommandBuffer:descriptor:aliasing** expects a valid Command Buffer to be passed in, the function didn't not mark the command buffer as an optional argument.
The reason for passing a command buffer is to allow MPS to reshape/materialize the memory if needed (thus encoding work to submit to the GPU on that command buffer). For view ops we shouldn't ever materialize the memory (as long as new new format can be represented using strides), so if MPS would try to make a copy for a view op, we would rather assert out so we catch the issue and fix it.
Note that I've run all torchbench networks / test_mps and test_modules and I haven't seen this issue.